vendor/engines/secure_rooms/db/migrate/20170405182534_add_tablet_token_to_card_readers.rb

```diff
@@ -0,0 +1,14 @@
+class AddTabletTokenToCardReaders < ActiveRecord::Migration
+
+  class SecureRooms::CardReader < ActiveRecord::Base
+  end
+
+  def change
+    add_column :secure_rooms_card_readers, :tablet_token, :string
+    add_index :secure_rooms_card_readers, :tablet_token, unique: true
+
+    SecureRooms::CardReader.reset_column_information
+    SecureRooms::CardReader.find_each { |cr| cr.update_column(:tablet_token, ("A".."Z").to_a.sample(12).join) }
```

Comment:
Conceivably, this could cause a uniqueness violation, but it seemed unlikely enough here, and it didn't seem worth bringing in the extra complexity I added to the main version of the model.